pkg/manager/internal.go

```diff
@@ -241,6 +249,19 @@ func (cm *controllerManager) GetHTTPClient() *http.Client {
 	return cm.cluster.GetHTTPClient()
 }
 
+// AddPrestartHook allows you to add prestart hooks.
+func (cm *controllerManager) AddPrestartHook(hook func(ctx context.Context) error) error {
```

Comment:
```suggestion
func (cm *controllerManager) AddHook(type HookType, fn Runnable) error {
```

Something like that? That would allow us to add more hook types in the future if we wanted to, also noticed that the signature is compatible with `Runnable`